components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/impl/MonitoringApiServiceImpl.java

```diff
@@ -157,108 +158,141 @@ public Response getAllWorkers(String username) throws NotFoundException {
-                                groupedWorkers.put(clusterID, workers);
-                            } else if (clusterID != null && (existing != null)) {
-                                existing.add(workerOverview);
-                            }
-                        } else {
-                            workerOverview.setWorkerId(worker.getWorkerId());
-                            ServerDetails serverDetails = new ServerDetails();
-                            serverDetails.setRunningStatus(Constants.NOT_REACHABLE_ID);
-                            workerOverview.setStatusMessage(getErrorMessage(workerResponse.status()));
-                            workerOverview.setServerDetails(serverDetails);
-                            workerOverview.setLastUpdate((long) 0);
-                            //grouping the never reached
-                            if (groupedWorkers.get(Constants.NEVER_REACHED) == null) {
-                                List<WorkerOverview> workers = new ArrayList<>();
-                                workers.add(workerOverview);
-                                groupedWorkers.put(Constants.NEVER_REACHED, workers);
-                            } else {
-                                List existing = groupedWorkers.get(Constants.NEVER_REACHED);
-                                existing.add(workerOverview);
+                        feign.Response resourceResponse = WorkerServiceFactory.getWorkerHttpsClient(
+                                PROTOCOL + generateURLHostPort(manager.getHost(), String.valueOf(manager
+                                        .getPort())), this.getUsername(), this.getPassword()).getClusterNodeDetails();
+                        if (resourceResponse.status() == 200) {
+                            Reader inputStream = resourceResponse.body().asReader();
+                            List<ResourceClusterInfo> clusterInfos = gson.fromJson(
+                                    inputStream, new TypeToken<List<ResourceClusterInfo>>() {}.getType());
+                            for (ResourceClusterInfo clusterInfo : clusterInfos) {
+                                String nodeId = clusterInfo.getNodeId();
+                                ResourceClusteredWorkerNode.add(nodeId);
                             }
                         }
                     } catch (feign.RetryableException e) {
-                        WorkerMetricsSnapshot lastSnapshot = WorkerStateHolder.getMetrics(worker.getWorkerId());
-                        if (lastSnapshot != null) {
-                            lastSnapshot.updateRunningStatus(Constants.NOT_REACHABLE_ID);
+                        if (logger.isDebugEnabled()) {
+                            logger.warn(removeCRLFCharacters(manager.getWorkerId()) + " Unnable to reach manager.", e);
+                        } else {
+                            logger.warn(removeCRLFCharacters(manager.getWorkerId()) + " Unnable to reach manager.");
```

Comment:
Shall we print this warn log even when debug is enabled? and print a debug log with the exception